src/cluster_legacy.c

```diff
@@ -5665,14 +5665,23 @@ void addNodeDetailsToShardReply(client *c, clusterNode *node) {
 /* Add the shard reply of a single shard based off the given primary node. */
 void addShardReplyForClusterShards(client *c, list *nodes) {
     serverAssert(listLength(nodes) > 0);
-    clusterNode *n = listNodeValue(listFirst(nodes));
     addReplyMapLen(c, 2);
     addReplyBulkCString(c, "slots");
 
-    /* Use slot_info_pairs from the primary only */
-    n = clusterNodeGetPrimary(n);
+    /* Find a node which has the slot information served by this shard.
+     * Note: `clusterGenNodesSlotsInfo` is invoked prior to this method call for pre-computation of slot_info_pairs
```

Comment:
I wonder if it is cleaner to just fold this function into `clusterCommandShards` then we don't need to explain the relationship between this function and `clusterGenNodesSlotsInfo` .

Not a hard blocker for me.

Reply:
This was the behavior prior to the change. Thought of adding a code comment for better readability.

Reply:
I understand this was the old code structure (and thanks for adding the comments, btw). I was thinking that, for someone who has recently joined the project, would it be easier for them if we just removed this function and folded the logic into the caller. Comments are important but it takes effort to keep them in sync and when they get out of sync, it becomes even more confusing for newcomers, which we do expect more with a project like Valkey.

Reply:
I agree it makes it cleaner. Have refactored in this https://github.com/valkey-io/valkey/pull/790/commits/aa003398aff9d1a7b61e5a8a1e35d3f250d8fe9c. PTAL.